examples/express/index.js

```diff
@@ -0,0 +1,159 @@
+const express = require('express')
+var session = require('express-session')
+
+const app = express()
+
+app.use(session({
+  secret: 'CHANGE-IT',
+  resave: false,
+  saveUninitialized: true,
+  cookie: {}
+}))
+
+// Import
+const helper = require('openid-client-helper')
+
+// // A ForgeRock configuration example.
+// const issuerMetadata = {
+//   issuer: 'https://default.iam.example.com/am/oauth2',
+//   revocation_endpoint: 'https://default.iam.example.com/am/oauth2/token/revoke' // A property not exposed in the OpenID configuration document.
+// }
+// const clientMetadata = {
+//   client_id: 'node-openid-client',
+//   client_secret: 'password',
+//   // token_endpoint_auth_method: 'client_secret_post', // The default is 'client_secret_basic'.
+//   redirect_uri: 'http://localhost:3333/redirect',
+//   post_logout_redirect_uri: 'http://localhost:3333'
+// }
+// const resources = {
+//   'https://default.iam.example.com/openidm/info/login': {
+//     scope: 'fr:idm:profile'
+//   }
+// }
+
+// // A Google configuration example.
+// const issuerMetadata = {
+//   issuer: 'https://accounts.google.com'
+// }
+// const clientMetadata = {
+//   client_id: 'google-client.apps.googleusercontent.com',
+//   client_secret: 'google-client-secret',
+//   // token_endpoint_auth_method: 'client_secret_post', // The default is 'client_secret_basic'.
+//   redirect_uri: 'http://localhost:3333/redirect'
+//   // `Google` does not support RP-initiated logout.
+// }
+// const resources = {
+//   'https://openidconnect.googleapis.com/v1/userinfo': {
+//     scope: 'https://www.googleapis.com/auth/userinfo.profile'
+//   }
+// }
+
+// Abstract configuration example.
+const issuerMetadata = {
+  issuer: 'openid-provider.well-known.openid-configuration-uri'
+}
+const clientMetadata = {
+  client_id: 'client-id',
+  client_secret: 'client-secret',
+  // token_endpoint_auth_method: 'client_secret_post', // The default is 'client_secret_basic'.
+  redirect_uri: 'http://localhost:3333/redirect',
+  post_logout_redirect_uri: 'http://localhost:3333'
+}
+const resources = {
+  'https://protected-resource-uri': {
+    scope: 'profile'
+  }
+}
+
+// Configure.
+const {
+  authorize,
+  redirect,
+  deauthorize,
+  getAuthorization,
+  fetch
+} = helper({
+  issuerMetadata,
+  clientMetadata,
+  resources
+})
+
+// Home.
+app.get('/', (req, res, next) => {
+  res.send('<a href="/authorize">Authorize</a>')
+})
+
+// Authorize.
+app.get('/authorize', authorize({
+  // An example of a special authorization parameter.
+  // In this case, it would instruct `Google` to issue a refresh token (for "offline" use).
+  access_type: 'offline'
+}))
+
+// Token request.
+app.get('/redirect', redirect(), (req, res) => {
+  res.redirect('/protected')
+})
+
+// Fetch.
+app.get('/protected', (req, res) => {
+  var content = '<p><a href="/">Home</a></p>'
+
+  content += '<p><a href="/deauthorize">Deauthorize</a></p>'
+
+  fetch(
+    Object.keys(resources)[0],
```

Comment:
It's not obvious to me that this is expected to be a URL (also it's randomly selected from the resources map, and the keys from the resources map aren't necessarily direct endpoints anyway - they are just the leading pattern). I think a more obvious example REST API URL would be better here (and I realize that it would make running this example slightly more involved, as it would have to be updated here as well, but that strikes me as an acceptable trade-off).

Reply:
Changed. I added a section in the example README as well.